### PR TITLE
Josie/advanced training

### PR DESCRIPTION
**Pull request by jtdkhamp**

### CODE REVIEW

File: Advanced-Recruit-Training/src/BatteryStateDisplayService.cpp [hunk 1]
```diff
@@ -29,5 +29,14 @@ void BatteryStateDisplayService::handleBatteryDataReceived(const BatteryData& ba
...
+    QString batTime = batteryStateOfChargeService_.timeWhenChargedOrDepleted().toString("hh:mm:ss.zzz");
+
+    if (batteryStateOfChargeService_.isCharging())
+    {
+        QTextStream(stdout) << "Time until full charge: " << batTime << endl;
+    }
+    else
+    {
+        QTextStream(stdout) << "Time until full depletion: " << batTime << endl;
+    }
```
Comment (louisj381) [inline Advanced-Recruit-Training/src/BatteryStateDisplayService.cpp]: I would recommend editing this function so the time to charge/depletion is displayed on the same line as the data, that way it's less confusing and easier to look at end result.

File: Advanced-Recruit-Training/src/BatteryStateOfChargeService.h [hunk 2]
```diff
@@ -24,5 +24,7 @@ class BatteryStateOfChargeService : public I_BatteryStateOfChargeService
     // Convention is to have an _ after private members
     const double initialStateOfChargePercent_;
 
-    // Add your necessary variables here
+    double current_;
```
Comment (louisj381) [inline Advanced-Recruit-Training/src/BatteryStateOfChargeService.h]: I would name this variable more explicitly, besides the first instance, this value is always a past value.

File: Advanced-Recruit-Training/src/BatteryStateOfChargeService.cpp [hunk 3]
```diff
@@ -23,50 +23,54 @@ double BatteryStateOfChargeService::totalAmpHoursUsed() const
...
 
 QTime BatteryStateOfChargeService::timeWhenChargedOrDepleted() const
 // calculates time (in hours) and returns it in QTime
 {
-    double time;
+    double hours;
     if (isCharging())
     {
-        time = ampHoursUsed_ / current_;
+        hours = ampHoursUsed_ / (currentCurrent_ * -1);
```
Comment (louisj381) [inline Advanced-Recruit-Training/src/BatteryStateOfChargeService.cpp]: There is a nice function that will take care of the negative one for you in QT documentation.

File: Advanced-Recruit-Training/src/BatteryStateOfChargeService.cpp [hunk 4]
```diff
@@ -23,50 +23,54 @@ double BatteryStateOfChargeService::totalAmpHoursUsed() const
...
     QTime nullTime(0, 0);
-    QTime clockTime = nullTime.addMSecs(time * MS_PER_HOUR);
+    QTime clockTime = nullTime.addMSecs(hours * MS_PER_HOUR);
     return clockTime;
 }
 
 void BatteryStateOfChargeService::addData(const BatteryData& batteryData)
 // update ampHoursUsed, time and current
 {
+    double previousCurrent = currentCurrent_;
```
Comment (louisj381) [inline Advanced-Recruit-Training/src/BatteryStateOfChargeService.cpp]: What is the previousCurrent defined as on the first set of data?

**Comment by bill-luu** [comment]
Looks good to me!

**Pull request closed without merging**